pkg/portableresources/backend/controller/createorupdateresource.go

```diff
@@ -76,6 +76,10 @@ func (c *CreateOrUpdateResource[P, T]) Run(ctx context.Context, req *ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	if data.ResourceMetadata().Status.Recipe == nil {
+		data.ResourceMetadata().Status.Recipe = &rpv1.RecipeStatus{}
+	}
```

Comment:
Since this is already being set in the processors, why do we need this additional check here?

Reply:
The processor sets the content of the pointer.
*v.Status = *output.Status

v.Status should not be nil here.